File: src/UVa/10812.cpp

/*
*****************************************************************
*                           PROBLEMA                            *
*****************************************************************
*                  UVa 10812 - Beat the Spread!                 *
*****************************************************************
*/

#include <cstdio>
#include <cmath>
using namespace std;

int main(){

	long long int n, s ,d;
	long long int x, y;

	scanf("%lld", &n);

	for(long long int i= 0; i<n; i++){

		scanf("%lld %lld", &s, &d);

		y = (s - d) / 2;
		x = s - y;

		if( x >= 0 && y >= 0 && (x + y == s) && (abs(x - y) == d))
			printf("%lld %lld\n", x, y);
		else
			printf("impossible\n");
			
	}

}

File: src/UVa/10812/10812.py
#!/usr/bin/env 

'''*
*****************************************************************
*                           PROBLEMA                            *
*****************************************************************
*                  UVa 10812 - Beat the Spread!                 *
*****************************************************************
*''' 

def main():
	
	n = int(input())

	for i in range(n):

		s, d = input().split()
		s, d = int(s), int(d)

		x = (d + s)//2
		y =  s - x

		if (x >= 0) and (y >= 0) and (x + y == s) and (abs(x - y) == d):
			print(x, y)
		else:
			print("impossible")

if __name__ == "__main__":
	main()
